rust/patchable/src/main.rs

```diff
@@ -263,8 +269,19 @@ fn main() -> Result<()> {
     .context(ConfigureGitLoggingSnafu)?;
 
     let opts = <Opts as clap::Parser>::parse();
-    let images_repo = repo::discover_images_repo(".").context(FindImagesRepoSnafu)?;
-    let images_repo_root = images_repo.workdir().context(NoImagesRepoWorkdirSnafu)?;
+    let images_repo_root_pathbuf = match opts.images_repo_root {
```

Comment:
nit:
```suggestion
    let images_repo_root = match opts.images_repo_root {
```
We can always borrow it where we need to.

Reply:
In fact, it would probably be cleaner to just move it into the PVC without borrowing:

```patch
diff --git a/rust/patchable/src/main.rs b/rust/patchable/src/main.rs
index d32f9b4b..c1adcb53 100644
--- a/rust/patchable/src/main.rs
+++ b/rust/patchable/src/main.rs
@@ -5,11 +5,7 @@ mod repo;
 mod utils;
 
 use core::str;
-use std::{
-    fs::File,
-    io::Write,
-    path::{Path, PathBuf},
-};
+use std::{fs::File, io::Write, path::PathBuf};
 
 use git2::{Oid, Repository};
 use serde::{Deserialize, Serialize};
@@ -35,12 +31,12 @@ struct ProductVersionConfig {
     base: Oid,
 }
 
-struct ProductVersionContext<'a> {
+struct ProductVersionContext {
     pv: ProductVersion,
-    images_repo_root: &'a Path,
+    images_repo_root: PathBuf,
 }
 
-impl ProductVersionContext<'_> {
+impl ProductVersionContext {
     fn load_config(&self) -> Result<ProductVersionConfig> {
         let path = &self.config_path();
         tracing::info!(
@@ -271,10 +267,12 @@ fn main() -> Result<()> {
         Some(path) => path,
         None => {
             let images_repo = repo::discover_images_repo(".").context(FindImagesRepoSnafu)?;
-            images_repo.workdir().context(NoImagesRepoWorkdirSnafu)?.to_owned()
+            images_repo
+                .workdir()
+                .context(NoImagesRepoWorkdirSnafu)?
+                .to_owned()
         }
     };
-    let images_repo_root = images_repo_root.as_path();
     match opts.cmd {
         Cmd::Checkout { pv, base_only } => {
             let ctx = ProductVersionContext {
```

Reply:
Yep, is changed now